test/e2e/framework/pod/wait.go

```diff
@@ -151,8 +193,8 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 		}
 
 		podList, err := c.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
+		lastAPIError = err
```

Comment:
is it possible to create a stack with all the api errors?

Reply:
Maybe it would be more useful to just log them as they occur? `handleWaitingAPIError` already does that, and I'll add an extra line to log non-retryable errors too.

Reply:
yeah, that sounds simpler and better, the error is logged with the timestamp and it will help to debug the failures better